code/parachain/frame/composable-traits/src/staking/reward/rate_based.rs

```diff
@@ -0,0 +1,99 @@
+use codec::{Decode, Encode, MaxEncodedLen};
```

Comment:
Let's not nest things unnecessarily without any of us having a deep understanding about how the protocol is going to evolve(We haven't launched).  As zen of python states, `Flat is better than nested`.

Reply:
I personally prefer to separate models (and their implementations) based on behaviour/ uses - they can always be reexported at the top level (which I should have done, not sure why I didn't :sweat_smile:) so changing the internal structure of the library/ module isn't a breaking change.

I can revert this to what it was before, with `lock` and `math` being the only submodules.

Reply:
yes, that preference is natural but that is a temptation we need to work against :) The problem is that it results in unnatural module boundaries/public interfaces when done without a deep understanding of the domain(not just in rust). Good refactorings are based on the business/technical merits rather than a shallow understanding of the current state (this is including me right now about staking).

Reply:
> I can revert this to what it was before, with lock and math being the only submodules.

`lock` and `math` are not supposed to be separate modules either. Just previous state of the code

Reply:
Fair enough, I'll revert the changes to what they were before and then I can remove the other submodules in a separate PR